auth_supplier/models/res_users.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    @api.model
+    def _signup_create_user(self, values):
+        res = super(ResUsers, self)._signup_create_user(values)
+        if isinstance(res, int):
+            user = self.env['res.users'].browse(res)
+            account_type = values.get('account_type', False)
+            if account_type and account_type == 'supplier':
```

Comment:
Shouldn't `account_type` be added as a field to this model?

Also, wouldn't it be better to add a `supplier` boolean instead? It could be a reference field to `partner_id.supplier` and the ORM would do the rest by itself, if I'm not wrong.